app/src/main/java/com/lastaoutdoor/lasta/ui/screen/social/components/AddFriendDialog.kt

```diff
@@ -62,43 +70,80 @@ fun AddFriendDialog(
                     LocalContext.current.getString(R.string.fr_email_add),
                     modifier = Modifier.testTag("SubHeader"),
                     textAlign = TextAlign.Center)
-                AddFriendForm(friendRequestFeedback, requestFriend)
+                AddFriendForm(
+                    friendRequestFeedback, friendSuggestions, requestFriend, fetchFriendSuggestions)
               }
         }
       }
 }
 
 @Composable
-private fun AddFriendForm(friendRequestFeedback: String, requestFriend: (String) -> Unit) {
+private fun AddFriendForm(
+    friendRequestFeedback: String,
+    friendSuggestions: List<UserModel>,
+    requestFriend: (String) -> Unit,
+    fetchFriendSuggestions: (String) -> Unit
+) {
 
   // to hide the keyboard
   val focusManager = LocalFocusManager.current
   // text inside the textfield
   var text by remember { mutableStateOf("") }
 
-  TextField(
-      value = text,
-      onValueChange = { text = it },
-      label = { Text(LocalContext.current.getString(R.string.email)) },
-      modifier = Modifier.fillMaxWidth().padding(8.dp).testTag("EmailTextField"),
-      keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
-      keyboardActions =
-          KeyboardActions(
-              onDone = {
-                focusManager.clearFocus()
-                requestFriend(text)
-              }))
+  Column(modifier = Modifier.fillMaxWidth(), horizontalAlignment = Alignment.CenterHorizontally) {
+    TextField(
+        value = text,
+        onValueChange = { newValue -> text = newValue },
+        label = { Text(LocalContext.current.getString(R.string.email)) },
+        modifier = Modifier.fillMaxWidth().padding(8.dp).testTag("EmailTextField"),
+        keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
+        keyboardActions =
+            KeyboardActions(
+                onDone = {
+                  focusManager.clearFocus()
+                  requestFriend(text)
+                }))
+
+    // Fetch friend suggestions when text changes
+    LaunchedEffect(text) {
+      // put text in small letters
+      fetchFriendSuggestions(text.lowercase())
+    }
+    // Display friend suggestions
+    LazyColumn(

```

Comment:
This is a very long composable, maybe you could extract a different one for the list to make it lighter ro read ?

Reply:
uWu it really does not seem really long to me, compared to others particularly